Formula/lean.rb

```diff
@@ -0,0 +1,35 @@
+class Lean < Formula
+  desc "Theorem prover"
+  homepage "https://leanprover.github.io/"
+  url "https://github.com/leanprover/lean/archive/v3.2.0.tar.gz"
+  sha256 "2668e075a9bdf4270d75fd60181078b33c078c7288b5bb8ad795819d8d5272cf"
+  head "https://github.com/leanprover/lean.git"
+
+  depends_on "cmake" => :build
+  depends_on "gmp"
+  depends_on "jemalloc" => :recommended
+  depends_on "gperftools" => :optional
+
+  def install
+    mkdir_p "build/release"
+    cd "build/release"
+    system "cmake", "../../src", *std_cmake_args
+    system "make", "install"
+  end
+
+  test do
+    (testpath/"hello.lean").write <<-EOS.undent
+      def id' {α : Type} (x : α) : α := x
+
+      inductive tree (α : Type) : Type
+      | node : α → list tree → tree
+
+      example (a b : Prop) : a ∧ b -> b ∧ a :=
+      begin
+          intro h, cases h,
+          split, repeat { assumption }
+      end
+    EOS
+    system "#{bin}/lean", testpath/"hello.lean"
```

Comment:
Test looks nice, but please check its output.

Reply:
There actually is no output; lean just verifies the input, which means it type checks the file and exits with a status of 0.